samples/snippets/src/test/java/com/example/spanner/admin/generated/SampleTestBaseV2.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.admin.generated;
+
+import com.example.spanner.SampleIdGenerator;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.admin.database.v1.DatabaseAdminClient;
+import com.google.cloud.spanner.admin.database.v1.DatabaseAdminSettings;
+import com.google.cloud.spanner.admin.instance.v1.InstanceAdminClient;
+import com.google.cloud.spanner.admin.instance.v1.InstanceAdminSettings;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+/**
+ * Base class for sample integration intests using auto-generated admin clients.
+ */
+public class SampleTestBaseV2 {
+
+  private static final String BASE_INSTANCE_ID =
+      System.getProperty("spanner.sample.instance", "mysample-instance");
+
+  private static final String BASE_DATABASE_ID =
+      System.getProperty("spanner.sample.database", "sampledb");
+  private static final String BASE_BACKUP_ID = "samplebk";
+  private static final String BASE_INSTANCE_CONFIG_ID = "sampleconfig";
+  private static final int AWAIT_TERMINATION_SECONDS = 10;
+
+  protected static String projectId;
+  protected static final String instanceId = System.getProperty("spanner.test.instance");
+  protected static DatabaseAdminClient databaseAdminClient;
+  protected static InstanceAdminClient instanceAdminClient;
+
+  protected static final String multiRegionalInstanceId =
+      System.getProperty("spanner.test.instance.mr");
+  protected static final String instanceConfigName = System
+      .getProperty("spanner.test.instance.config");
+  protected static SampleIdGenerator idGenerator;
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    final String serverUrl = "";
+    final SpannerOptions.Builder optionsBuilder =
+        SpannerOptions.newBuilder().setAutoThrottleAdministrativeRequests();
+    final SpannerOptions options = optionsBuilder.build();
+    final DatabaseAdminSettings.Builder databaseAdminSettingsBuilder =
+        DatabaseAdminSettings.newBuilder();
+    final InstanceAdminSettings.Builder instanceAdminSettingBuilder =
+        InstanceAdminSettings.newBuilder();
+
+    if (!serverUrl.isEmpty()) {
```

Comment:
nit: In the current setup, this will always be false (the first line of  this method reads `final String serverUrl = ""`. Should we make `serverUrl` configurable?

Reply:
This is intentional. I had refactored this bit so that its easy for people to override the server URL when they want to run test again a non-production server URL. In that case they would just set final String serverUrl = "SOME_VALID_URL". We instruct devs to do the same in a few of our internal dev guides.